### ISSUE TITLE
Feat: support for customize paste function

### ISSUE THREAD

**Issue by CatKevin** [singerdmx/flutter-quill#2156]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/singerdmx/flutter-quill/issues)

### Use case

I copied some content from another software that may contain some parsable emoji, but when I paste them into Quill Editor they can't parse the emoji using the custom parsing method.

### Proposal

I found that there are onImagePaste and onGifPaste methods in QuillEditorConfigurations to handle the problem of pasting pictures. I wonder if there should be an onPaste method to handle the pasted content?

```
Future<String?> Function(Uint8List)? onImagePaste,
  Future<String?> Function(Uint8List)? onGifPaste,
```

**Comment by EchoEllet** [comment]
Thank you for the report, do you want to customize how paste is handled in the editor in general? or do something to paste an image?

**Comment by CatKevin** [comment]
> Thank you for the report, do you want to customize how paste is handled in the editor in general? or do something to paste an image?

Yes! I want to customize how paste is handled in the editor in general! Because I want to handle some specific encoding formats, such as email, url, emoji and other formats! So that we can make full use of the powerful functions of flutter quill! ! I Hope this function can be added! Thank you very much! !

**Comment by EchoEllet** [comment]
Once our platform clipboard implementation becomes stable (still in early stage) and web common issues fixed, we will start working on providing more API to customize the behavior and extend it further.

**Comment by xalanq** [comment]
Any progress?

**Comment by EchoEllet** [comment]
> Any progress?

Fixed in the pre-release version (`11.0.0`). See [this comment](https://github.com/singerdmx/flutter-quill/issues/2350#issuecomment-2464796551). Not closed yet since I'm not entirely sure what is expected to be customized, and also it's a pre-release, expect some minor changes (will be documented in `CHANGELOG.md`).

**Comment by EchoEllet** [comment]
@CatKevin Can you confirm whether the `onDeltaPaste` and `onPlainTextPaste` fixes the issue? https://github.com/singerdmx/flutter-quill/issues/2350#issuecomment-2464796551

**Comment by CatKevin** [comment]
Dear Ellet,


Thank you for your reply!
However, I only see string2string and delta2delta related conversions. But we may copy a string with emojis somewhere (not in quill), such as "Hello[smile]Mike![cheer][cheer][cheer]", and users can copy and paste it on any APP without exposing the emoji protocol in this APP. After we paste this string in quill, we actually hope to parse the strings "[smile]" and "[cheer]" into corresponding emojis and display them in quill.
Therefore, I think we may also need a method to convert strings to Delta. Do you think so?



https://github.com/singerdmx/flutter-quill/issues/2350#issuecomment-2514020932



Looking forward to your reply!


Best,
Kevin


| |
Kevin
|
|
***@***.***
|


---- Replied Message ----
| From | ***@***.***> |
| Date | 12/2/2024 21:30 |
| To | ***@***.***> |
| Cc | ***@***.***> ,
***@***.***> |
| Subject | Re: [singerdmx/flutter-quill] Feat: support for customize paste function (Issue #2156) |

@CatKevin Can you confirm whether the onDeltaPaste and onPlainTextPaste fixes the issue? #2350 (comment)

**Comment by EchoEllet** [comment]
Thanks for the suggestion, see https://github.com/singerdmx/flutter-quill/issues/2350#issuecomment-2514128816
